### DISCUSSION TITLE
Static check custom Printf-like functions

### DISCUSSION THREAD

**Discussion by paukul** [dominikh/go-tools#1345] [Q&A]
Is there a way to add a linter for functions that accept printf-like arguments?
let's assume I have the following function:
```
func foo(some int, another *thing, format string, a ...any) {
 // do stuff with `some` and `another`
 fmt.Printf(format, a...)
}
```
I'd like staticcheck to complain about missing/missmatched `a` params, just like it does for `Printf`, `Sprintf` and the likes.

Either by adding an own check or by configuring it somehow to treat the function like `Printf` for the last arugments?
Cheers

**Comment by dominikh** [comment]
I'll think about a way of adding configuration for this.

**Reply by zozobreakzou** [reply]
Has this been supported? like the go vet -printfuncs
